design/merge-patch-and-strategic-in-resource-modifier.md

```diff
@@ -68,17 +68,12 @@ resourceModifierRules:
     namespaces:
     - ns1
   mergePatches:
-  - patchData: |
-      {
-        "metadata": {
-          "annotations": {
-            "foo": null
-          }
-        }
-      }
+  - patchData:
+      metadata:
+        annotations:
+          foo: null
 ```
 - The above configmap will apply the Merge Patch to all the pods in namespace ns1 and remove the annotation `foo` from the pods.
-- Both json and yaml format are supported for the patchData.
```

Comment:
oh what caused this to change?
I thought we discussed this support in the library ?

Reply:
We should somehow support JSON since it's a jsonpatch based support and people would prefer to use their kubectl json setup to here as is.

Reply:
Both json and yaml format are supported inside the patch logic, but there is an issue: we can not mix json and yaml in one menifest. So if the RM configMap is json, patchData here can also be json.
I can update the design to support json format for RM configmap, what do you think?

Reply:
Didn't fully understand what you meant by this...

Reply:
which means if the ResourceModifiers is in yaml format, patchData must be yaml too, if the ResourceModifiers is json, patchData must be json.

but there is an exception:
if you build the ResourceModifiers configMap programmatically, patchData can be either json or yaml, because we you call `yaml.Marshal` to convert ResourceModifiers from go object to yaml, patchData will always be convert to yaml. Also, if you call `json.Marshal` to convert ResourceModifiers from go object to json, patchData will always be convert to json.

Reply:
Replied to below comment in continuation to this.

Reply:
But Velero doesn't support `ResourceModifier` in JSON format at this moment, right?

I guess we cannot use the `yaml.Unmarshal()` functions directly in order to support both JSON and YAML formats, we need to do some decoding/encoding by ourselves according to different formats before calling `yaml.Unmarshal()`.

Reply:
thank to @anshulahuja98 I changed the type of patchData to string, and now we can support both json and yaml patchData